internal/gengapic/paging_test.go

```diff
@@ -216,3 +248,244 @@ func TestIterTypeOf(t *testing.T) {
+				Type:     typep(descriptor.FieldDescriptorProto_TYPE_MESSAGE),
+				Label:    labelp(descriptor.FieldDescriptorProto_LABEL_REPEATED),
+				TypeName: proto.String(".diregapic.paging.ItemsEntry"),
+			},
+		},
+	}
+
+	// Methods
+	validPageSizeMthd := &descriptor.MethodDescriptorProto{
+		Name:       proto.String("ValidPageSize"),
+		InputType:  proto.String(".diregapic.paging.ValidPageSizeRequest"),
+		OutputType: proto.String(".diregapic.paging.ValidRepeatedResponse"),
+	}
+	validMaxResultsMthd := &descriptor.MethodDescriptorProto{
+		Name:       proto.String("ValidMaxResults"),
+		InputType:  proto.String(".diregapic.paging.ValidMaxResultsRequest"),
+		OutputType: proto.String(".diregapic.paging.ValidMapResponse"),
+	}
+	clientStreamingMthd := &descriptor.MethodDescriptorProto{
+		Name:            proto.String("ClientStreaming"),
+		InputType:       proto.String(".diregapic.paging.ValidPageSizeRequest"),
+		OutputType:      proto.String(".diregapic.paging.ValidRepeatedResponse"),
+		ClientStreaming: proto.Bool(bool(true)),
+	}
+	serverStreamingMthd := &descriptor.MethodDescriptorProto{
+		Name:            proto.String("ServerStreaming"),
+		InputType:       proto.String(".diregapic.paging.ValidPageSizeRequest"),
+		OutputType:      proto.String(".diregapic.paging.ValidRepeatedResponse"),
+		ServerStreaming: proto.Bool(bool(true)),
+	}
+	tooManyRepeatedMthd := &descriptor.MethodDescriptorProto{
+		Name:       proto.String("TooManyRepeated"),
+		InputType:  proto.String(".diregapic.paging.ValidPageSizeRequest"),
+		OutputType: proto.String(".diregapic.paging.TooManyRepeatedResponse"),
+	}
+	tooManyMapMthd := &descriptor.MethodDescriptorProto{
+		Name:       proto.String("TooManyMap"),
+		InputType:  proto.String(".diregapic.paging.ValidPageSizeRequest"),
+		OutputType: proto.String(".diregapic.paging.TooManyMapResponse"),
+	}
+
+	file := &descriptor.FileDescriptorProto{
+		Package: proto.String("diregapic.paging"),
+		Options: &descriptor.FileOptions{
+			GoPackage: proto.String("mypackage"),
+		},
+		MessageType: []*descriptor.DescriptorProto{
+			invalidRsp,
+			mapEntry,
+			randomMessage,
+			tooManyMap,
+			tooManyRepeated,
+			validMap,
+			validMaxResults,
+			validPageSize,
+			validRepeated,
+		},
+		Service: []*descriptor.ServiceDescriptorProto{
+			&descriptor.ServiceDescriptorProto{
+				Name: proto.String("TestService"),
+				Method: []*descriptor.MethodDescriptorProto{
+					clientStreamingMthd,
+					serverStreamingMthd,
+					tooManyMapMthd,
+					tooManyRepeatedMthd,
+					validMaxResultsMthd,
+					validPageSizeMthd,
+				},
+			},
+		},
+	}
+
+	req := plugin.CodeGeneratorRequest{
+		Parameter: proto.String("go-gapic-package=path;mypackage,transport=rest"),
+		ProtoFile: []*descriptor.FileDescriptorProto{file},
+	}
+	g.init(&req)
+
+	for _, tst := range []struct {
+		mthd      *descriptor.MethodDescriptorProto
+		sizeField *descriptor.FieldDescriptorProto // A nil field means this is not a Diregapic paged method
```

Comment:
The `diregapicPagingField` function will return non-nils even  for correct but non-DIREGAPIC paging, IIUC?

Reply:
Correct. My original thought was to check for regular paging first then diregapic paging. Since it's a superset, however, it makes sense to just have the one function.